wrapspawner/wrapspawner.py

```diff
@@ -129,6 +129,14 @@ def poll(self):
         else:
             return _yield_val(1)
 
+    if hasattr(Spawner, 'progress'):
```

Comment:
I think it's fine to proxy the `progress` method unconditionally. This will only ever be called if the Jupyterhub version supports it, and if so, the child spawner class is guaranteed to have at least a fallback `.progress` method due to inheriting from the base `Spawner` class.

Reply:
I went back and forth on the behavior several times, too.  In theory we shouldn't have to worry about failures.  But suppose someone is using an interactive debugger or uses a `hasattr(cls, 'progress')`.  An AttributeError makes sense if you try to access something that doesn't exist, but here we are sort of asserting that it exists.  Mainy since this is a `@property` and not function call, even accessing it, possibly other autocompletions and who knows what else, sees something that exists and then gives AttributeError when you try to access it.

There were enough possible special cases because it was `@property` that I finally thought it simpler to condition the whole thing.  Slightly uglier in a readable way, but many fewer possible surprises.